src/librustc_typeck/check/mod.rs

```diff
@@ -2997,27 +2997,22 @@ fn check_expr_with_unifier<'a, 'tcx, F>(fcx: &FnCtxt<'a, 'tcx>,
                                  skip : Vec<InternedString>) {
         let name = field.node.as_str();
         // only find fits with at least one matching letter
-        let mut best_dist = name.len();
-        let mut best = None;
-        for elem in &variant.fields {
-            let n = elem.name.as_str();
-            // ignore already set fields
-            if skip.iter().any(|x| *x == n) {
-                continue;
-            }
-            // ignore private fields from non-local crates
-            if variant.did.krate != LOCAL_CRATE && elem.vis != Visibility::Public {
-                continue;
-            }
-            let dist = lev_distance(&n, &name);
-            if dist < best_dist {
-                best = Some(n);
-                best_dist = dist;
-            }
-        }
-        if let Some(n) = best {
+        let best_dist = Some(name.len());
```

Comment:
I'm fine with this. Be sure to update the doc comment explaining the third argument.

I don't think the `best_dist` variable is necessary here, just put the `Some(name.len())` inline in the call to find_best_match.